MoTrPAC/motrpac-frontend
# Connects to #135. QC Data Monitor migration from ShinyApp.

**Technical Notes:**
* In this initial version of the migration, the data source are static JSON files derived from the pipeline-generated `.txt` files.
* The JSON data files are, for now, embedded in the app rather than making them publicly accessible via Google Drive or GCS bucket.
* In the attempt to mirror ShinyApp's functionalities, there are further UI modifications given the feedback received upon reviews and testing.

**Steps to test:**
1. Sign into the data hub portal
2. In the sidebar, click on the **QC Data Monitor** button link and expect the **PASS1B** tissue submission "grid/heatmap" to be shown.
3. Click through the phase tabs (e.g. PASS1A/1C, Human) and expect to see respective grids representing the phases.
4. Now click on the **Metabolomics** button in the upper-right and expect to see the sample data table to be shown.
5. Verify the sorting, filtering, paging, and navigating functionalities against the data in the table.
6. Repeat the above step for other data tables (**Proteomics, RNA-seq,** etc.).
7. Now logout and sign in again as a registered external user.
8. Upon signing in, expect to see the **QC Data Monitor** button link disabled in the sidebar.
